src/Microsoft.Framework.Configuration/ConfigurationSection.cs

```diff
@@ -22,7 +24,8 @@ public ConfigurationSection(IList<IConfigurationSource> sources)
         {
             get
             {
-                return Get(key);
```

Comment:
Indexer implementation should live in common base class and do all the right argument checks.